android/src/main/java/com/polidea/flutter_ble_lib/converter/ScanResultJsonConverter.java

```diff
@@ -0,0 +1,92 @@
+package com.polidea.flutter_ble_lib.converter;
+
+import android.support.annotation.NonNull;
+
+import com.polidea.multiplatformbleadapter.AdvertisementData;
+import com.polidea.multiplatformbleadapter.ScanResult;
+import com.polidea.multiplatformbleadapter.utils.Base64Converter;
+import com.polidea.multiplatformbleadapter.utils.UUIDConverter;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+public class ScanResultJsonConverter implements JsonConverter<ScanResult> {
+
+    private interface Metadata {
```

Comment:
Do all those data (rssi, mtu, etc.) are connected only with the scan result?

Reply:
To be honest `mtu` should be associated only with the connection. Not with ScanResult

Reply:
We don't have a `Connection` object. Best we could do is put it in the device, though I suppose it's unknown on iOS and unknown until a request to change it on Android?

Reply:
This contract was copy-pasted from the JS code base. Do we want to change it now?

Reply:
@pawelByszewski AFAIK rssi should be contained only in ScanResult because we can read it's real value only during scanning operation.

Reply:
I submitted an issue to remove MTU from ScanResult class in MBA.